mill-scalafix/src/com/goyeau/mill/scalafix/ScalafixCache.scala

```diff
@@ -0,0 +1,24 @@
+package com.goyeau.mill.scalafix
+
+import com.goyeau.mill.scalafix.CoursierUtils
+import coursier.core.Repository
+import scalafix.interfaces.Scalafix
+
+import java.util.concurrent.ConcurrentHashMap
+import scala.jdk.CollectionConverters._
+import scala.ref.SoftReference
+
+private[scalafix] object ScalafixCache {
+
+  private val cache = new ConcurrentHashMap[(String, Seq[Repository]), SoftReference[Scalafix]]
+
+  def getOrElseCreate(scalaVersion: String, repositories: Seq[Repository]) =
+    cache.get((scalaVersion, repositories)) match {
```

Comment:
Having a blocking method like `computeIfAbsent` would avoid getting several concurrent initialisations of the same instance on projects with many modules. This helped with sbt, not sure if the threading model is the same with mill.

Reply:
That's a great idea!
I used `compute` since I wanted to keep the `SoftReference`.